src/RequestPool.cpp

```diff
@@ -75,6 +75,48 @@ auto RequestPool::Produce(
         return Request(this, std::move(request_handle_ptr));
     }
 }
+auto RequestPool::Produce(
```

Comment:
OK so it seems like you need to know max_download_bytes when producing since it affects init() and what CURLOPTs get set on the curl handle.  So I would say use an ssize_t and set it to -1 by default, which means "download the entire file", then you do not need another overload on the Produce() function but just an additional defaulted parameter

Reply:
That makes a lot more sense!